### PR TITLE
Feature/create space

### LINKED ISSUE #8: As a proprietor, I want to be able to create and modify my spaces. (epic-account-page)

### PR DESCRIPTION
**Pull request by bishalr1**
#8  The current stable version of creating and updating spaces. 
Includes initial testing for form, model, and database accuracy.

### CODE REVIEW

**Review by sharletclaros — DISMISSED** [review]
Some newlines needed at the end of files. Syntax changes can be done after all merging is complete, so lgtm!

File: .gitignore
Comment (jimmy-coleman) [inline .gitignore]: Apologies if I missed a discussion on this, but shouldn't we commit `/migrations` so that our database schema is up to date with our merges? I understand we may have to do some coordination, but I think we should consider removing this from the .gitignore.

**Review by btruong9 — DISMISSED** [review]
For best practices, styling in templates can be moved to a separate file. This can be done later, so it looks good to me!

File: README.md
Comment (jimmy-coleman) [inline README.md]: 'likely due to selenium and less due to the webapp' sounds a bit shakey, and may make people going through our README less confident in our code. I believe we discussed holding off on Selenium to make sure it's testing isn't integrated until it is working, so maybe we should update our README to reflect that.

File: requirements.txt
Comment (jimmy-coleman) [inline requirements.txt]: I think we should have requirements as >= rather than strictly ==; if we update Django, for example some minor patch, our website won't be compatible to run.

File: sadcowpng/sharedspaces/forms.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/forms.py]: super super nit: My old host told me that when commenting, you don't need to mention the data structure explicitly. Instead, saying that this represents a multiple choice field mapping for question x is a bit better. Again, super nit but I always thought it was a nice thing to have in mind to write clearer comments.

File: sadcowpng/sharedspaces/forms.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/forms.py]: This comment mentions 'Might'. I see that we're still using BooleanFields here. Make sure that this final decision is reflected in the comment, as I'm not sure if you switched from something else to boolean fields for clarity or vice versa.

File: sadcowpng/sharedspaces/forms.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/forms.py:24]: We don't want to merge commented code into our main and develop branch. All merged code should serve a purpose.

File: sadcowpng/sharedspaces/models.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/models.py:15]: Same as earlier comment, we don't want to merge commented code. All code should serve a purpose

File: sadcowpng/sharedspaces/models.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/models.py]: nit: 'sent to space' is a bit vague.

File: sadcowpng/sharedspaces/models.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/models.py]: Again, merged comments should reflect a decision that was taken, not reflect a consideration that you were making while developing.

File: sadcowpng/sharedspaces/templates/sharedspaces/create_space.html
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/templates/sharedspaces/create_space.html]: Consider breaking up this line into multiple lines for readability, perhaps by html field arguments

File: sadcowpng/sharedspaces/templates/sharedspaces/create_space.html
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/templates/sharedspaces/create_space.html]: Irrelevant for this merge, but I wanted to write it down that in the future we definitely want to consider centralizing styles. That said, that's a later front-end story issue.

File: sadcowpng/sharedspaces/templates/sharedspaces/update_space.html
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/templates/sharedspaces/update_space.html]: nit: not properly indented

File: sadcowpng/sharedspaces/tests.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/tests.py]: I've noticed that you've repeated this same block of comments. If that's the case, maybe you should take it out of both locations and instead write about it in either the README or above the class by describing the testing flow.

File: sadcowpng/sharedspaces/tests.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/tests.py]: repeated again, so I definitely think we should consider maybe centralizing the documentation.

File: sadcowpng/sharedspaces/views.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/views.py]: nit: This import comes from the same module as the import above, so you could put it on the same line. I think that's technically best practice, along with alphabetizing imports but again I have all of GeoDjango to learn tonight so I don't have room to nitpick too much.

File: sadcowpng/sharedspaces/views.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/views.py:26]: Is this something that still needs to be done? Again, comments should always reflect the final decision that was made, not uncertainty

**Review by jimmy-coleman — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (bishalr1) [reply .gitignore]: Yep, thank you for pointing it out.

File: README.md
Comment (bishalr1) [reply README.md]: yep forgot to remove thank you

File: requirements.txt
Comment (bishalr1) [reply requirements.txt]: yep will update that

File: sadcowpng/sharedspaces/forms.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/forms.py]: Updated

File: sadcowpng/sharedspaces/forms.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/forms.py:24]: keeping it for later development so for now we will keep it.

File: sadcowpng/sharedspaces/models.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/models.py:15]: Will keep it for later development. will remove in future tho

File: sadcowpng/sharedspaces/forms.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/forms.py]: removed

File: sadcowpng/sharedspaces/models.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/models.py]: Fixed

File: sadcowpng/sharedspaces/models.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/models.py]: Hahaha, that actually funny. But yes I was saying we need to send our data to space. Clarified the statement.

File: sadcowpng/sharedspaces/views.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/views.py:26]: That will be added later in the next iteration.

File: sadcowpng/sharedspaces/templates/sharedspaces/create_space.html
Comment (bishalr1) [reply sadcowpng/sharedspaces/templates/sharedspaces/create_space.html]: for now, we did not have a styles.css file so just adding it in the HTML file and moving it when we get all the basic web pages set up.

File: sadcowpng/sharedspaces/templates/sharedspaces/update_space.html
Comment (bishalr1) [reply sadcowpng/sharedspaces/templates/sharedspaces/update_space.html]: Fixed

File: sadcowpng/sharedspaces/templates/sharedspaces/create_space.html
Comment (bishalr1) [reply sadcowpng/sharedspaces/templates/sharedspaces/create_space.html]: updated

File: sadcowpng/sharedspaces/tests.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/tests.py]: Removed to a central location at the beginning of all database accuracy test.

File: sadcowpng/sharedspaces/tests.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/tests.py]: fixed

File: sadcowpng/sharedspaces/views.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/views.py]: Condensed it down a bit more.

File: sadcowpng/sharedspaces/forms.py
Comment (jimmy-coleman) [inline sadcowpng/sharedspaces/forms.py:13]: super super super nit: 'for for'

**Review by jimmy-coleman — APPROVED** [review]
Great job, lgtm!

File: sadcowpng/sharedspaces/forms.py
Comment (bishalr1) [reply sadcowpng/sharedspaces/forms.py:13]: Whoops got it

**Pull request merged**
